### DISCUSSION TITLE
[General] How to increase time with hardhat testing

### DISCUSSION THREAD

**Discussion by rafflero** [zkSync-Community-Hub/zksync-developers#102] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I would like to increase the time, similar to the hardhat toolbox or OZ helpers whereby I can either set the desired time by passing a timestamp, or increasing the time in desired increments. This sort of testing is needed for my contracts to see if they properly function.

### What is the specific issue or error you're encountering?

I don't know how to increase the time on the `era_test_node`. I have looked through the repo for the node and couldn't find anything there as well, nor anything in the zk docs.

### Can you share the error messages or logs you're receiving, if any?

-

### Have you made any recent changes to the contract before encountering this issue?

-

### Are there any external libraries or contracts that your contract interacts with?

-

### Can you provide the relevant portions of your contract code where the issue is occurring?

-

### Have you tried to isolate the problem, and if so, what were the results?

-

### What steps have you already taken to try to resolve the issue?

-

**Comment by uF4No** [comment, marked as answer]
Hi @rafflero 

That option is not available yet, but it's one of the next priorities. We've an [open issue](https://github.com/matter-labs/era-test-node/issues/29) for it

**Reply by rafflero** [reply]
Thank you for your fast response @uF4No!

I will leave this ticket open for others to find until the functionality is implemented.

Do you have an ETA for the functionality?

**Reply by MexicanAce** [reply]
No ETA at the moment, but others can follow along in that Issue for the latest information. For now, we can mark this Discussion as answered.

**Comment by bradfordbradv** [comment]
To increase time for testing purposes in the era_test_node, you can utilize the evm_increaseTime JSON-RPC method. This method allows you to increase the timestamp of the next mined block by the specified number of seconds. Here's an example of how to use it: {
  "jsonrpc": "2.0",
  "method": "evm_increaseTime",
  "params": [3600], // Increase time by 1 hour (3600 seconds)
  "id": 12345
 You can send this JSON-RPC request to your era_test_node endpoint using tools like cURL or a JSON-RPC client library. This will effectively simulate the passage of time in your test environment, allowing you to test time-dependent functionality in your contracts.

**Comment by bartlettdv** [comment]
Environment: Testnet
Functionality: Implement time manipulation for contract testing purposes.
Issue: Unable to find a method to increase time on the era_test_node.
Potential Solution: Explore available options within the era_test_node or zkSync documentation for time manipulation.

**Comment by lupanncoofonpowell** [comment]
Currently, there isn't a built-in function or tool within zkSync's ecosystem to manipulate time for testing purposes. However, you can explore alternative testing frameworks or custom solutions to simulate time-based scenarios in your contracts.

**Comment by spatounmas** [comment]
Increasing time during testing on zkSync's testnet can be achieved by using the advanceTimeAndBlock function provided by Hardhat or similar testing frameworks. However, it seems that there is no built-in functionality for time manipulation in the zkSync test node itself. You may need to simulate time manipulation locally in your testing environment using libraries like Hardhat's time manipulation tools or by manually adjusting timestamps in your test scripts.

**Comment by dianocsemintrey** [comment]
The user is trying to implement a functionality to increase the time on the era_test_node on the zkSync Testnet. The user is looking for a way to set the desired time by passing a timestamp or increasing the time in desired increments, similar to the hardhat toolbox or OZ helpers. However, the user couldn't find any information on how to do this in the repo for the node or in the zkSync documentation. No specific error messages or logs are provided, and no recent changes have been made to the contract. No relevant portions of the contract code are provided, and no steps have been taken to isolate the problem. No repo link is provided.
